cmd/artifact/main_test.go

```diff
@@ -0,0 +1,328 @@
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	"github.com/urfave/cli"
+)
+
+type logger struct {
+	t *testing.T
+	n string
+}
+
+func (l *logger) Write(p []byte) (n int, err error) {
+	l.t.Logf("STDOUT: %s", p)
+	return len(p), nil
+}
+
+type testEnv struct {
+	taskID         string
+	runID          string
+	inputFilename  string
+	outputFilename string
+	queue          *queue.Queue
+	t              *testing.T
+}
+
+func (e testEnv) validate() {
+	f1, err := ioutil.ReadFile(e.inputFilename)
+	if err != nil {
+		e.t.Fatal(err)
+	}
+	f2, err := ioutil.ReadFile(e.outputFilename)
+	if err != nil {
+		e.t.Fatal(err)
+	}
+	if !bytes.Equal(f1, f2) {
+		e.t.Error("File %s and %s unexpectedly differ", e.inputFilename, e.outputFilename)
+	}
+}
+
+func setup(t *testing.T) (testEnv, func()) {
+	tEnv := testEnv{}
+	taskID := slugid.Nice()
+	taskGroupID := slugid.Nice()
+	tEnv.taskID = taskID
+	tEnv.t = t
+
+	input, err := ioutil.TempFile(".", "test-file-input")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tEnv.inputFilename = input.Name()
+	buf := make([]byte, 1024*1024)
+	for i := 0; i < 10; i++ {
+		_, err := rand.Read(buf)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = input.Write(buf)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	input.Sync()
+	err = input.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	output, err := ioutil.TempFile(".", "test-file-output")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tEnv.outputFilename = output.Name()
+	output.Sync()
+	err = output.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// This command creates a task that has a deadline in 15 minutes
+	created := time.Now().UTC()
+	// reset nanoseconds
+	created = created.Add(time.Nanosecond * time.Duration(created.Nanosecond()*-1))
+	// deadline in one hour' time
+	deadline := created.Add(15 * time.Minute)
+	// expiry in one day, in case we need test results
+	expires := created.AddDate(0, 0, 2)
+
+	taskDef := &queue.TaskDefinitionRequest{
+		Created:      tcclient.Time(created),
+		Deadline:     tcclient.Time(deadline),
+		Expires:      tcclient.Time(expires),
+		Extra:        json.RawMessage(`{}`),
+		Dependencies: []string{},
+		Requires:     "all-completed",
+		Metadata: struct {
+			Description string `json:"description"`
+			Name        string `json:"name"`
+			Owner       string `json:"owner"`
+			Source      string `json:"source"`
+		}{
+			Description: "taskcluster-lib-artifact-go test",
+			Name:        "taskcluster-lib-artifact-go test",
+			Owner:       "taskcluster-lib-artifact-go-ci@mozilla.com",
+			Source:      "https://github.com/taskcluster/taskcluster-lib-artifact-go",
+		},
+		Payload:       json.RawMessage(`{}`),
+		ProvisionerID: "no-provisioner",
+		Retries:       1,
+		Routes:        []string{},
+		SchedulerID:   "test-scheduler",
+		Scopes:        []string{},
+		Tags:          json.RawMessage(`{"CI":"taskcluster-lib-artifact-go/cli"}`),
+		Priority:      "lowest",
+		TaskGroupID:   taskGroupID,
+		WorkerType:    "my-workertype",
+	}
```

Comment:
it looks like it might.... if we use specific names, but I'd like to use descriptive names and not overwrite artifacts in the test.  I'll look into this as part of enabling travis or taskcluster on this repository.